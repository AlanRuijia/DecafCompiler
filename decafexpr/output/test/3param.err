output prefix: /Users/Jerry/Develop/SFU/compiler/DecafCompiler/decafexpr/output/test/llvm/3param
llvmas: /usr/local/Cellar/llvm38/3.8.0/lib/llvm-3.8/bin/llvm-as
llc: /usr/local/Cellar/llvm38/3.8.0/lib/llvm-3.8/bin/llc
cc: gcc
codegen: /Users/Jerry/Develop/SFU/compiler/DecafCompiler/decafexpr/answer/decafexpr
stdlib: /Users/Jerry/Develop/SFU/compiler/DecafCompiler/decafexpr/answer/decaf-stdlib.c
generating llvm code:/Users/Jerry/Develop/SFU/compiler/DecafCompiler/decafexpr/answer/decafexpr ... ok
; ModuleID = 'C'

define i32 @main() {
entry:
  %foo = alloca i32
  %main = alloca i32
}

define i32 @foo(i32, i32, i1) {
entry:
}

define i32 @main.1() {
entry:
  ret i32 0
}
assembling to bitcode:/usr/local/Cellar/llvm38/3.8.0/lib/llvm-3.8/bin/llvm-as "/Users/Jerry/Develop/SFU/compiler/DecafCompiler/decafexpr/output/test/llvm/3param.llvm" -o "/Users/Jerry/Develop/SFU/compiler/DecafCompiler/decafexpr/output/test/llvm/3param.llvm.bc" ... failed (1)
/usr/local/Cellar/llvm38/3.8.0/lib/llvm-3.8/bin/llvm-as: /Users/Jerry/Develop/SFU/compiler/DecafCompiler/decafexpr/output/test/llvm/3param.llvm:7:1: error: expected instruction opcode
}
^
converting to native code:/usr/local/Cellar/llvm38/3.8.0/lib/llvm-3.8/bin/llc "/Users/Jerry/Develop/SFU/compiler/DecafCompiler/decafexpr/output/test/llvm/3param.llvm.bc" -o "/Users/Jerry/Develop/SFU/compiler/DecafCompiler/decafexpr/output/test/llvm/3param.llvm.s" ... ok
linking:gcc -o "/Users/Jerry/Develop/SFU/compiler/DecafCompiler/decafexpr/output/test/llvm/3param.llvm.exec" "/Users/Jerry/Develop/SFU/compiler/DecafCompiler/decafexpr/output/test/llvm/3param.llvm.s" "/Users/Jerry/Develop/SFU/compiler/DecafCompiler/decafexpr/answer/decaf-stdlib.c" ... ok
running:/Users/Jerry/Develop/SFU/compiler/DecafCompiler/decafexpr/output/test/llvm/3param.llvm.exec ... ok
