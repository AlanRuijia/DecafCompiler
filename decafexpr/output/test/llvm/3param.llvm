; ModuleID = 'C'

declare void @print_int(i32)

define i32 @foo(i32, i32, i1) {
entry:
}

define i32 @main() {
entry:
  ret i32 0
}
