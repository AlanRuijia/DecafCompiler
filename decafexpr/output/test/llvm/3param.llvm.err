; ModuleID = 'C'

declare void @print_int(i32)

define i32 @foo(i32, i32, i1) {
entry:
  store i32 1, i32* %x
  store i32 1, i32* %y
  store i1 false, i1* %z
  %x = load i32, i32* %x
  call void @print_int(i32 %x)
  %y = load i32, i32* %y
  call void @print_int(i32 %y)
  %z = load i1, i1* %z
  %zexttmp = zext i1 %z to i32
  call void @print_int(i32 %zexttmp)
  store i32 %0, i32* %x
  store i32 %1, i32* %y
  store i1 %2, i1* %z
  ret i32 0
}

define i32 @main() {
entry:
  %calltmp = call i32 @foo(i32 1, i32 2, i1 true)
  ret i32 0
}
