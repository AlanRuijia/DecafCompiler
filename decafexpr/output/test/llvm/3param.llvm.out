Debug message: Generating extern function...
Arg Type: 17 
Inserting method decl into symbol table...
Inserting block into symbol table...
Inserting method decl into symbol table...
Inserting block into symbol table...
Debug message: Generating program...
Debug message: Generating package...
Debug message: Generating FieldDecls...
Debug message: Generating MethodDecl...
Defind variable in line 3 : x
Codegen for Assign to x
Defind variable in line 3 : y
Codegen for Assign to y
Defind variable in line 3 : z
Codegen for Assign to z
Debug message: Generating method call... Calling print_int
